lib/internal/test_runner/test.js

```diff
@@ -409,17 +409,26 @@ class Test extends AsyncResource {
   }
 
   matchesTestNamePatterns() {
-    const matchesByNameOrParent = ArrayPrototypeSome(testNamePatterns, (re) =>
+    let patterns;
+    let bool = true;
```

Comment:
Please don't call variables by their type but by their function :) In this case boolean is the type, the function is to determine whether or not we're filtering by name pattern or skipPattern.

The code can simplified to:

```js
const patterns = testNamePatterns ?? testSkipPatterns;
const isSkipping = !testNamePatterns; // name takes precedence over skip
const matchesByNameOrParent = ArrayPrototypeSome(patterns, (re) =>
  RegExpPrototypeExec(re, this.name) !== null,
) || this.parent?.matchesTestNamePatterns();
if (matchesByNameOrParent) {
  return !isSkipping;  
}  
// ....
```